src/tx_pool_monitor.ts

```diff
@@ -0,0 +1,115 @@
+        this._signatureBySignatureHash = {};
+        for (const functionAbi of _.filter(contractAbi, {type: 'function', constant: false})) {
+            const types = _.map(functionAbi.inputs, input => input.type);
+            const humanReadableTypes = _.map(functionAbi.inputs, input => `${input.type} ${input.name}`);
+            const signature = functionAbi.name + '(' + types.join(',') + ')';
+            const humanReadableSignature = functionAbi.name + '(' + humanReadableTypes.join(',') + ')';
+            const signatureHash = this._web3Wrapper.sha3(signature).substr(0, 10);
+            this._typesBySignatureHash[signatureHash] = types;
+            this._signatureBySignatureHash[signatureHash] = humanReadableSignature;
+        }
+    }
+    public watch() {
+        this.watchBlocks();
+        this.watchMempool();
+    }
+    private watchBlocks(): void {
+        const config = {};
+        const blockAndLogStreamer = new BlockAndLogStreamer(
+            this._web3Wrapper.getBlockByHashAsync.bind(this._web3Wrapper),
+            _.noop as any,
+            config,
+        );
+        const getLatestBlock = this._web3Wrapper.getLatestBlockAsync.bind(this._web3Wrapper);
+        const onBlockAddedSubscriptionToken = blockAndLogStreamer.subscribeToOnBlockAdded(
+            this.onBlockAdded.bind(this));
+        const onBlockRemovedSubscriptionToken = blockAndLogStreamer.subscribeToOnBlockRemoved(
+            this.onBlockRemoved.bind(this));
+        this._blockWatchIntervalId = setInterval(async () => {
+            blockAndLogStreamer.reconcileNewBlock(await getLatestBlock());
+        }, MONITORING_INTERVAL_MS);
+    }
+    private async onBlockAdded(block: any): Promise<void> {
+        console.log('Block added');
+        for (const tx of block.transactions) {
+            if (tx.to !== CONTRACT_ADDRESS) {
+                continue;
+            }
+            if (_.isUndefined(this._state[tx.hash])) {
+                console.log(`Out of a mempool - https://etherscan.io/tx/${tx.hash}`);
+            } else {
+                console.log(`Mempool + https://etherscan.io/tx/${tx.hash}`);
+                delete this._state[tx.hash];
+            }
+        }
+    }
+    private async onBlockRemoved(block: any): Promise<void> {
+        console.log('Block removed');
+        for (const tx of block.transactions) {
+            if (tx.to !== CONTRACT_ADDRESS) {
+                continue;
+            }
+            if (_.isUndefined(this._state[tx.hash])) {
+                console.log(`Adding back to the mempool - https://etherscan.io/tx/${tx.hash}`);
+                this._state[tx.hash] = tx;
+            } else {
+                console.log(`Removed transaction that was already in the mempool  https://etherscan.io/tx/${tx.hash}`);
+            }
+        }
+    }
+    private watchMempool(): void {
+        this._txPoolWatchIntervalId = setInterval(this.stepAsync.bind(this), MONITORING_INTERVAL_MS);
```

Comment:
Make sure you call `clearInterval` on this in a `destroy` method that then gets called in the 0x.ts `destroy` method.